src/main/java/ClientHandler.java

```diff
@@ -11,17 +12,25 @@ public ClientHandler(Socket socket) {
     }
 
     public void run() {
-        System.out.println("Client Connected.");
         try {
             out = new PrintWriter(new OutputStreamWriter(clientSocket.getOutputStream()), true);
             in = new BufferedReader(new InputStreamReader(clientSocket.getInputStream()));
-            out.println("Welcome");
-            String input;
-            while ((input = in.readLine()) != null) {
-                if(input != ".") {
-                    out.println("Hey.");
-                }
+
+            String line;
+            ArrayList<String> input = new ArrayList<>();
+
+            while ((line = in.readLine()) != null && line.length() != 0) {
+                input.add(line);
             }
+
+            String parameters = RequestReader.requestHandler(input);
```

Comment:
Would be interested to hear more about this approach in IPM. Can you tell us more about what type of method this is and it's use?